File: src/main/java/um/tesoreria/rest/model/Legajo.java
/**
 * 
 */
package um.tesoreria.rest.model;

import java.io.Serializable;
import java.math.BigDecimal;
import java.time.OffsetDateTime;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.Table;
import jakarta.persistence.UniqueConstraint;

import com.fasterxml.jackson.annotation.JsonFormat;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.EqualsAndHashCode;
import lombok.NoArgsConstructor;

/**
 * @author daniel
 *
 */
@Data
@Entity
@Table(name = "aluleg", uniqueConstraints = {
		@UniqueConstraint(columnNames = { "ale_fac_id", "ale_per_id", "ale_doc_id" }) })
@EqualsAndHashCode(callSuper = false)
@NoArgsConstructor
@AllArgsConstructor
public class Legajo extends Auditable implements Serializable {
	/**
	 * 
	 */
	private static final long serialVersionUID = -3950450930864967230L;

	@Id
	@GeneratedValue(strategy = GenerationType.IDENTITY)
	@Column(name = "ale_id")
	private Long legajoId;

	@Column(name = "ale_per_id")
	private BigDecimal personaId;

	@Column(name = "ale_doc_id")
	private Integer documentoId;

	@Column(name = "ale_fac_id")
	private Integer facultadId;

	@Column(name = "ale_leg_id")
	private Long numeroLegajo = 0L;

	@Column(name = "ale_fecha")
	@JsonFormat(shape = JsonFormat.Shape.STRING, pattern = "yyyy-MM-dd'T'HH:mm:ssZ", timezone = "UTC")
	private OffsetDateTime fecha;

	@Column(name = "ale_lec_id")
	private Integer lectivoId;

	@Column(name = "ale_pla_id")
	private Integer planId;

	@Column(name = "ale_car_id")
	private Integer carreraId;

	@Column(name = "ale_carrera")
	private Byte tieneCarrera = 0;

	@Column(name = "ale_geo_id")
	private Integer geograficaId;

	@Column(name = "ale_contrasenia")
	private String contrasenha;

	@Column(name = "intercambio")
	private Byte intercambio = 0;

}

File: src/main/java/um/tesoreria/core/kotlin/model/Legajo.kt
package um.tesoreria.core.kotlin.model

import java.math.BigDecimal
import java.time.OffsetDateTime
import jakarta.persistence.Column
import jakarta.persistence.Entity
import jakarta.persistence.GeneratedValue
import jakarta.persistence.GenerationType
import jakarta.persistence.Id
import jakarta.persistence.Table
import jakarta.persistence.UniqueConstraint
import com.fasterxml.jackson.annotation.JsonFormat
import jakarta.persistence.JoinColumn
import jakarta.persistence.JoinColumns
import jakarta.persistence.OneToOne

@Entity
@Table(
    name = "aluleg",
    uniqueConstraints = [UniqueConstraint(columnNames = ["ale_fac_id", "ale_per_id", "ale_doc_id"])]
)
data class Legajo(

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "ale_id")
    var legajoId: Long? = null,

    @Column(name = "ale_per_id")
    var personaId: BigDecimal? = null,

    @Column(name = "ale_doc_id")
    var documentoId: Int? = null,

    @Column(name = "ale_fac_id")
    var facultadId: Int? = null,

    @Column(name = "ale_leg_id")
    var numeroLegajo: Long = 0L,

    @Column(name = "ale_fecha")
    @JsonFormat(shape = JsonFormat.Shape.STRING, pattern = "yyyy-MM-dd'T'HH:mm:ssZ", timezone = "UTC")
    var fecha: OffsetDateTime? = null,

    @Column(name = "ale_lec_id")
    var lectivoId: Int? = null,

    @Column(name = "ale_pla_id")
    var planId: Int? = null,

    @Column(name = "ale_car_id")
    var carreraId: Int? = null,

    @Column(name = "ale_carrera")
    var tieneCarrera: Byte = 0,

    @Column(name = "ale_geo_id")
    var geograficaId: Int? = null,

    @Column(name = "ale_contrasenia")
    var contrasenha: String? = null,

    @Column(name = "intercambio")
    var intercambio: Byte = 0,

    @OneToOne(optional = true)
    @JoinColumns(
        JoinColumn(name = "ale_fac_id", referencedColumnName = "car_fac_id", insertable = false, updatable = false),
        JoinColumn(name = "ale_pla_id", referencedColumnName = "car_pla_id", insertable = false, updatable = false),
        JoinColumn(name = "ale_car_id", referencedColumnName = "car_id", insertable = false, updatable = false)
    )
    var carrera: Carrera? = null

) : Auditable()
